### ISSUE TITLE
Laravel route:cache

### ISSUE THREAD

**Issue by EgorGruzdev** [sebastiaanluca/laravel-module-loader#3]
It may be worth adding a check for the existence of a route cache.

/src/Providers/Provider.php
```
    /**
     * Map out all predefined module routes.
     */
    protected function mapRoutes() : void
    {
        if(false === $this->app->routesAreCached()) {
            foreach ($this->routers as $router) {
                $this->app->make($router);
            }
        }
    }
```
https://voltagead.com/laravel-route-caching-for-improved-performance/

**Comment by sebastiaanluca** [comment]
Good catch! Must've overlooked this. I'll try to add it this week and make sure it's tested too.

**Comment by EgorGruzdev** [comment]
When is it planned to add this feature to the release?
I also suggest adding verification for configuration files.

\SebastiaanLuca\Module\Providers\ModuleProvider::registerConfiguration:

```php
     /**
     * Merge the default module configuration file with its published counterpart.
     *
     * @return void
     */
    protected function registerConfiguration() : void
    {
        if($this->app->configurationIsCached()) return;

        $name = $this->getLowercasePackageName();

        $configuration = "{$this->getModulePath()}/config/{$name}.php";

        if (! file_exists($configuration)) {
            return;
        }

        $this->mergeConfigFrom(
            $configuration,
            $name
        );
    }
```

**Comment by sebastiaanluca** [comment]
I've just released v2 which includes the route caching, configuration caching, and the providers registration array. Also a few additional tests and some fixes. Sorry it took a while, completely lost track of this. Was going to add more tests, but that will be for later.

Regarding the configuration caching: I've added it as it might mean a small speed improvement, but in essence it doesn't do much. If a configuration file was cached and then a module merges its own config with the cached version, nothing will be overwritten or added as the cached version already contained all the keys and values. But now it'll bypass the file check and `require`, so it's something.

Thanks for the suggestions! Let me know if you have any issues regarding these updates.
